modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -139,8 +139,32 @@ class GoogleCloudStorageBlobStore implements BlobStore {
         this.casBackoffPolicy = casBackoffPolicy;
     }
 
-    private MeteredStorage client() throws IOException {
-        return storageService.client(projectId, clientName, repositoryName, statsCollector);
+    /**
+     * Get a client that will retry according to its configured settings
+     *
+     * @return A client
+     */
+    MeteredStorage client() throws IOException {
+        return storageService.client(
+            projectId,
+            clientName,
+            repositoryName,
+            statsCollector,
+            GoogleCloudStorageService.RetryBehaviour.ClientConfigured
+        );
+    }
+
+    /**
+     * Get a client that will not retry on failure
+     *
+     * @return A client with max retries configured to zero
+     */
+    MeteredStorage clientNoRetries() throws IOException {
+        return storageService.client(projectId, clientName, repositoryName, statsCollector, GoogleCloudStorageService.RetryBehaviour.None);
+    }
+
+    int getMaxRetries() {
+        return storageService.clientSettings(projectId, clientName).getMaxRetries();
     }
```

Comment:
In GCS, retries are configured at the client level, not the request level. So we need to create a separate client that's configured to not retry, so we can manage retries in the `RetryingInputStream`. This will mean an extra client is cached for each config, but I don't think that's a huge overhead.